chalicelib/utils.py

```diff
@@ -58,21 +59,22 @@ def workspace_exists(self):
             s.mount('https://', HTTPAdapter(
                 max_retries=Retry(total=4,
                                   backoff_factor=1,
-                                  redirect=False,
-                                  history=(),
-                                  raise_on_status=True,
+                                  raise_on_status=False,
                                   status_forcelist=[401, 500, 502, 503, 504])
             ))
-            return s.get(url=url, headers=headers)
-        except:
-            return Response(body={},
-                            headers={'Content-Type': 'application/json'},
-                            status_code=500)
+            resp = s.get(url=url, headers=headers)
+            resp.raise_for_status()
```

Comment:
You do not need the `try` any more, and should not call `resp.raise_for_status()`, you can just return the response directly.

Also, based on our earlier discussions, I thought you said you always wanted to return a Chalice response object from methods in utils.py; you are now returning a requests.Response object.

Reply:
sorry, I was mistaken. the truth is that I always return a requests response object to shove data back and forth from app.py to utils.py, except in cases of failure. this is a special case because we're passing it to another method, regardless of success or failure.